EventGenerator/src/SingleProcessGenerator_module.cc

```diff
@@ -96,14 +101,19 @@ namespace mu2e {
     auto output{std::make_unique<StageParticleCollection>()};
 
     const auto simh = event.getValidHandle<SimParticleCollection>(simsToken_);
-    const auto mus = stoppedMuMinusList(simh);
-
-    for(const auto& mustop: mus) {
+    auto mus_temp = stoppedMuMinusList(simh); //default, negative muon stops
+ 
+    if (pdgId_==-11){
+      mus_temp = stoppedMuPlusList(simh); //positive muon stops
+      muonLifeTime_=0; //decay time already included for stopped muon(+)
```

Comment:
Can you add a "FIXME" comment here since we will want to change this when the stopped mu+ time no longer includes the decay time?

Reply:
Added FIXME.